ocsp/responder/multi_source.go

```diff
@@ -60,7 +60,15 @@ func (src *multiSource) Response(ctx context.Context, req *ocsp.Request) (*Respo
 	// from reaching the backend layer (Redis) and causing connections to be closed
 	// unnecessarily.
 	// https://blog.uptrace.dev/posts/go-context-timeout.html
-	secondaryChan := getResponse(context.Background(), src.secondary, req)
+	redisCtx := context.Background()
+	cancel := func() {}
+	deadline, ok := ctx.Deadline()
+	if ok {
+		redisCtx, cancel = context.WithDeadline(redisCtx, deadline)
+	}
+	defer cancel()
```

Comment:
Doesn't this still result in the redisCtx being cancelled as soon as this function returns (i.e. immediately after the primary source returns, if it isn't stale), resulting in lots of connection churn? I think we just want
```suggestion
	deadline, ok := ctx.Deadline()
	if ok {
		redisCtx, _ = context.WithDeadline(redisCtx, deadline)
	}
```

Reply:
Good catch. I've gotten used to thinking you have to always call the cancel func, but [here's what the docs actually say](https://pkg.go.dev/context):

>  Failing to call the CancelFunc leaks the child and its children until the parent is canceled _or the timer fires_. The go vet tool checks that CancelFuncs are used on all control-flow paths.

(emphasis mine)

So the contexts will eventually get cleaned up, but they'll wait for the deadline (typically 4.9s in our configs).